k8s/migration/internal/controller/rdmdisk_controller.go

```diff
@@ -0,0 +1,347 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v3/volumes"
+	vjailbreakv1alpha1 "github.com/platform9/vjailbreak/k8s/migration/api/v1alpha1"
+	constants "github.com/platform9/vjailbreak/k8s/migration/pkg/constants"
+	utils "github.com/platform9/vjailbreak/k8s/migration/pkg/utils"
+)
+
+// RDMDiskReconciler reconciles a RDMDisk object
+type RDMDiskReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=vjailbreak.k8s.pf9.io,resources=rdmdisks,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=vjailbreak.k8s.pf9.io,resources=rdmdisks/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=vjailbreak.k8s.pf9.io,resources=rdmdisks/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the RDMDisk object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.20.4/pkg/reconcile
+func (r *RDMDiskReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logf.FromContext(ctx)
+
+	// Get the RDMDisk resource
+	rdmDisk := &vjailbreakv1alpha1.RDMDisk{}
+	if err := r.Get(ctx, req.NamespacedName, rdmDisk); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			log.Error(err, "unable to fetch RDMDisk")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// Updated logic for handling phases and conditions
+	switch rdmDisk.Status.Phase {
+	case "":
+		var mostRecentValidationFailedCondition *metav1.Condition
+		for i := range rdmDisk.Status.Conditions {
+			cond := rdmDisk.Status.Conditions[i]
+			if cond.Type == "ValidationFailed" && cond.Status == metav1.ConditionTrue {
+				if mostRecentValidationFailedCondition == nil || cond.LastTransitionTime.After(mostRecentValidationFailedCondition.LastTransitionTime.Time) {
+					mostRecentValidationFailedCondition = &cond
+				}
+			}
+		}
+		if mostRecentValidationFailedCondition != nil {
+			lastTransitionTime := mostRecentValidationFailedCondition.LastTransitionTime.Time
+			if time.Since(lastTransitionTime) < 2*time.Minute {
+				log.Info("Skipping validation as a ValidationFailed condition was set less than 2 minutes ago", "RDMDisk", rdmDisk.Name, "LastReason", mostRecentValidationFailedCondition.Reason)
+				// Requeue after the remaining time to reach 2 minutes
+				requeueAfter := 2*time.Minute - time.Since(lastTransitionTime)
+				return ctrl.Result{RequeueAfter: requeueAfter}, nil
+			}
+		}
```

Comment:
I guess this whole block can also be inside above block where we set `mostRecentValidationFailedCondition = &cond` 

Not sure about readability/amount of nesting with that though. I will leave it upto you.